File: app/src/main/java/com/example/movies/core/helper/DateFormatHelper.java
package com.example.movies.core.helper;

import android.annotation.SuppressLint;

import com.example.movies.core.util.Constants;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

@SuppressLint("SimpleDateFormat")
public class DateFormatHelper {

    public static String formatDate(String date){
        SimpleDateFormat spf = new SimpleDateFormat(Constants.UNFORMATTED_DATE);
        try {
            Date newDate = spf.parse(date);
            spf = new SimpleDateFormat(Constants.FORMATTED_DATE);
            return spf.format(newDate);
        } catch (ParseException e) {
           return date;
        }
    }
}

File: app/src/main/java/com/example/movies/core/helper/DateFormatHelper.kt
package com.example.movies.core.helper

import android.annotation.SuppressLint

import com.example.movies.core.util.Constants

import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.Date

@SuppressLint("SimpleDateFormat")
object DateFormatHelper {

    fun formatDate(date: String): String {
        var spf = SimpleDateFormat(Constants.UNFORMATTED_DATE)
        try {
            val newDate = spf.parse(date)
            spf = SimpleDateFormat(Constants.FORMATTED_DATE)
            return spf.format(newDate)
        } catch (e: ParseException) {
            return date
        }

    }
}
